commit f9f57da2c2d119dbf109e3f6e1ceab7659294046
Author: Juergen Gross <jgross@suse.com>
Date:   Thu Feb 9 08:22:17 2023 +0100

    x86/mtrr: Revert 90b926e68f50 ("x86/pat: Fix pat_x_mtrr_type() for MTRR disabled case")
    
    Commit
    
      90b926e68f50 ("x86/pat: Fix pat_x_mtrr_type() for MTRR disabled case")

  crogers1:
    This commit also happens to set PAGE_CACHE_MODE_UC_MINUS on the e820 reserved
    regions for certain dom0 resources. This causes ioremap errors when drivers
    try to map these regions, such as tpm_tis, which ultimately fails to load the
    driver.  This commit was merged upstream to address an issue with USB disk
    enclosures originally.

--- a/arch/x86/mm/pat/memtype.c
+++ b/arch/x86/mm/pat/memtype.c
@@ -434,8 +434,7 @@ static unsigned long pat_x_mtrr_type(u64
 		u8 mtrr_type, uniform;
 
 		mtrr_type = mtrr_type_lookup(start, end, &uniform);
-		if (mtrr_type != MTRR_TYPE_WRBACK &&
-		    mtrr_type != MTRR_TYPE_INVALID)
+		if (mtrr_type != MTRR_TYPE_WRBACK)
 			return _PAGE_CACHE_MODE_UC_MINUS;
 
 		return _PAGE_CACHE_MODE_WB;
